### PR TITLE
LOG_X replaced with an spdlog

### PR DESCRIPTION
**Pull request by T-Britton**
### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x ] New feature (spdlog)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
no, down the road supplying a logger level may be needed,,,but this maintains compatibility
### Does this PR change default behavior?
no

### CODE REVIEW

File: src/algorithms/calorimetry/CalorimeterClusterMerger.cc [hunk 1]
```diff
@@ -16,8 +16,13 @@ using namespace dd4hep;
 //------------------------
 // AlgorithmInit
 //------------------------
-void CalorimeterClusterMerger::AlgorithmInit() {
+void CalorimeterClusterMerger::AlgorithmInit(spdlog::level::level_enum loglevel) {
+    m_logger->set_level(loglevel);
```
Comment (DraTeots) [inline src/algorithms/calorimetry/CalorimeterClusterMerger.cc:20]: Log level should be set from Factory init ans is taken from users parameter. Example: 

```c++
// Get log level from user parameter or default
    std::string log_level_str = "info";
    pm->SetDefaultParameter("Tracking:RawHitReco:LogLevel", log_level_str, "verbosity: trace, debug, info, warn, err, critical, off");
    m_log->set_level(eicrecon::ParseLogLevel(log_level_str));
```

Also note this
https://github.com/eic/EICrecon/discussions/68

and review of logger init below

File: src/algorithms/calorimetry/CalorimeterClusterMerger.cc [hunk 2]
```diff
@@ -39,31 +44,32 @@ void CalorimeterClusterMerger::AlgorithmProcess() {
...
     // loop over all position clusters and match with energy clusters
-    if (false) {
-      LOG_INFO(default_cout_logger) << "Step 1/1: Merging clusters where needed" << LOG_END;
+    if (m_logger->level() == spdlog::level::debug) {
+       m_logger->debug( "Step 1/1: Merging clusters where needed" );
     }
     for (const auto& [mcID, clusters] : clusterMap) {
-      if (false) {
-        LOG_INFO(default_cout_logger) << " --> Processing " << clusters.size() << " clusters for mcID " << mcID << LOG_END;
+      if (m_logger->level() == spdlog::level::debug) {
```
Comment (DraTeots) [inline src/algorithms/calorimetry/CalorimeterClusterMerger.cc:66]: First you don't need 

```c++
if(m_logger->level() == something)
```
if you do only `m_logger->debug(`

but the most importantly, if you do comparison, compare like this: 

```c++
if (m_logger->level() <= spdlog::level::debug) {
```
**<=** - so this `if` is true when log level debug OR trace

**Review by DraTeots — CHANGES_REQUESTED** [review]
Need some changes in how logger is instantiated and used

File: src/algorithms/calorimetry/CalorimeterClusterMerger.cc [hunk 3]
```diff
@@ -39,31 +39,32 @@ void CalorimeterClusterMerger::AlgorithmProcess() {
     auto& assoc2 = m_outputAssociations;
 
     if (!split.size()) {
-      if (false) {
-        LOG_INFO(default_cout_logger) << "Nothing to do for this event, returning..." << LOG_END;
+      if (m_logger->level() <=spdlog::level::debug) {
```
Comment (DraTeots) [inline src/algorithms/calorimetry/CalorimeterClusterMerger.cc:42]: This comparison is not needed if you just do `m_logger->debug(...)`. But in terms of functioning, this is good now

File: src/algorithms/calorimetry/CalorimeterClusterMerger.cc [hunk 4]
```diff
@@ -39,31 +39,32 @@ void CalorimeterClusterMerger::AlgorithmProcess() {
...
+       m_logger->debug( "Step 0/1: Getting indexed list of clusters..." );
+      //LOG_INFO(default_cout_logger) << "Step 0/1: Getting indexed list of clusters..." << LOG_END;
     }
     // get an indexed map of all vectors of clusters, indexed by mcID
     auto clusterMap = indexedClusterLists(split, assoc);
 
     // loop over all position clusters and match with energy clusters
-    if (false) {
-      LOG_INFO(default_cout_logger) << "Step 1/1: Merging clusters where needed" << LOG_END;
+    if (m_logger->level() <=spdlog::level::debug) {
```
Comment (DraTeots) [inline src/algorithms/calorimetry/CalorimeterClusterMerger.cc:57]: `If` could be removed

File: src/algorithms/calorimetry/CalorimeterClusterMerger.cc [hunk 5]
```diff
@@ -39,31 +39,32 @@ void CalorimeterClusterMerger::AlgorithmProcess() {
...
+       m_logger->debug( "Step 1/1: Merging clusters where needed" );
     }
     for (const auto& [mcID, clusters] : clusterMap) {
-      if (false) {
-        LOG_INFO(default_cout_logger) << " --> Processing " << clusters.size() << " clusters for mcID " << mcID << LOG_END;
+      if (m_logger->level() <=spdlog::level::debug) {
```
Comment (DraTeots) [inline src/algorithms/calorimetry/CalorimeterClusterMerger.cc:61]: `If` could be removed

File: src/algorithms/calorimetry/CalorimeterClusterMerger.cc [hunk 6]
```diff
@@ -86,8 +87,8 @@ void CalorimeterClusterMerger::AlgorithmProcess() {
         int nhits = 0;
         auto position = new_clus->getPosition();
         for (auto clus : clusters) {
-          if (false) {
-            LOG_INFO(default_cout_logger) << "   --> Adding cluster with energy: " << clus->getEnergy() << LOG_END;
+          if (m_logger->level() <=spdlog::level::debug) {
```
Comment (DraTeots) [inline src/algorithms/calorimetry/CalorimeterClusterMerger.cc:90]: `If` could be removed

File: src/algorithms/calorimetry/CalorimeterClusterMerger.cc [hunk 7]
```diff
@@ -104,8 +105,8 @@ void CalorimeterClusterMerger::AlgorithmProcess() {
         new_clus->setTime(time / energy);
         new_clus->setNhits(nhits);
         new_clus->setPosition(position / energy);
-        if (false) {
-          LOG_INFO(default_cout_logger) << "   --> Merged cluster with energy: " << new_clus->getEnergy() << LOG_END;
+        if (m_logger->level() <=spdlog::level::debug) {
```
Comment (DraTeots) [inline src/algorithms/calorimetry/CalorimeterClusterMerger.cc:108]: `If` could be removed

File: src/algorithms/calorimetry/CalorimeterClusterMerger.h [hunk 8]
```diff
@@ -64,15 +68,17 @@ class CalorimeterClusterMerger {
           break;
         }
       }
+      
       //TODO:spdlog verbosity
-      if (false) {
-        LOG_INFO(default_cout_logger) << " --> Found cluster with mcID " << mcID << " and energy "
-                  << cluster->getEnergy() << LOG_END;
+      if ( m_logger->level() == spdlog::level::debug) {
```
Comment (DraTeots) [inline src/algorithms/calorimetry/CalorimeterClusterMerger.h:73]: **<=**

File: src/algorithms/calorimetry/CalorimeterClusterMerger.h [hunk 9]
```diff
@@ -64,15 +68,17 @@ class CalorimeterClusterMerger {
...
+        m_logger->debug("--> Cluster {} has MC ID {} and energy", cluster->id(), mcID, cluster->getEnergy());
+        //LOG_INFO(default_cout_logger) << " --> Found cluster with mcID " << mcID << " and energy " << cluster->getEnergy() << LOG_END;
       }
 
       if (mcID < 0) {
-        if (false) {
-          LOG_INFO(default_cout_logger) << "   --> WARNING: no valid MC truth link found, skipping cluster..." << LOG_END;
+        if (m_logger->level() == spdlog::level::debug) {
```
Comment (DraTeots) [inline src/algorithms/calorimetry/CalorimeterClusterMerger.h:79]: **<=**

Comment (DraTeots) [reply src/algorithms/calorimetry/CalorimeterClusterMerger.h:79]: And if could be skipped at all

File: src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc [hunk 10]
```diff
@@ -67,9 +68,9 @@ void CalorimeterClusterRecoCoG::AlgorithmProcess() {
     for (const auto& pcl : proto) {
       auto cl = reconstruct(pcl);
 
-      if (false) {
-        LOG_INFO(default_cout_logger) << cl.getNhits() << " hits: " << cl.getEnergy() / GeV << " GeV, (" << cl.getPosition().x / mm << ", "
-                << cl.getPosition().y / mm << ", " << cl.getPosition().z / mm << ")" << LOG_END;
+      if (m_logger->level() == spdlog::level::debug) {
```
Comment (DraTeots) [inline src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc:71]: If could be skipped but if now **<=**

File: src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc [hunk 11]
```diff
@@ -141,8 +150,9 @@ void CalorimeterClusterRecoCoG::AlgorithmProcess() {
         edm4eic::MCRecoClusterParticleAssociation* cassoc = new edm4eic::MCRecoClusterParticleAssociation(*clusterassoc);
         m_outputAssociations.push_back(cassoc);
       } else {
-        if (false) {
-          LOG_INFO(default_cout_logger) << "No mcHitCollection was provided, so no truth association will be performed." << LOG_END;
+        if (m_logger->level() == spdlog::level::debug) {
```
Comment (DraTeots) [inline src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc:153]: It is just a 1/3 of changes... *<=*

File: src/algorithms/calorimetry/CalorimeterIslandCluster.h [hunk 12]
```diff
@@ -224,9 +228,9 @@ class CalorimeterIslandCluster {
                    std::vector<edm4eic::ProtoCluster *>& proto) const {
     // special cases
     if (maxima.empty()) {
-      if (false){//msgLevel(MSG::VERBOSE)) {
-        LOG_TRACE(default_cout_logger) << "No maxima found, not building any clusters" << LOG_END;
-
+      if (m_logger->level() >= spdlog::level::info){//msgLevel(MSG::VERBOSE)) {
```
Comment (DraTeots) [inline src/algorithms/calorimetry/CalorimeterIslandCluster.h:231]: ! probably <= here

Comment (DraTeots) [reply src/algorithms/calorimetry/CalorimeterIslandCluster.h:231]: And it looks like `if` is not needed here

**Review by DraTeots — CHANGES_REQUESTED** [review]
There are couple of places which needs to be fixed

File: src/algorithms/calorimetry/CalorimeterClusterMerger.cc (hunk 3 above)
Comment (T-Britton) [reply src/algorithms/calorimetry/CalorimeterClusterMerger.cc:42]: desire to leave the code as close to the original for now

File: src/algorithms/calorimetry/CalorimeterClusterMerger.cc (hunk 4 above)
Comment (T-Britton) [reply src/algorithms/calorimetry/CalorimeterClusterMerger.cc:57]: desire to leave as close to the original as possible, for now

File: src/algorithms/calorimetry/CalorimeterClusterMerger.cc (hunk 6 above)
Comment (T-Britton) [reply src/algorithms/calorimetry/CalorimeterClusterMerger.cc:90]: leaving as close to the original as possible

File: src/algorithms/calorimetry/CalorimeterClusterMerger.cc (hunk 7 above)
Comment (T-Britton) [reply src/algorithms/calorimetry/CalorimeterClusterMerger.cc:108]: leaving as close to the original as possible

File: src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc (hunk 10 above)
Comment (T-Britton) [reply src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc:71]: fixed in incoming push

File: src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc (hunk 11 above)
Comment (T-Britton) [reply src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc:153]: fixed (neglected in my find and replace) ...in incoming push

**Review by DraTeots — APPROVED** [review]

**Pull request merged**